### PR TITLE
feat(): stop old, deprecated builds

### PR DESCRIPTION
**Pull request by Izak88**

### CODE REVIEW

File: src/api/process-manager.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { startDockerImageSetupJob, startBuildProcess, stopContainer } from './process';
 import { Observable, Subject, BehaviorSubject, Subscription } from 'rxjs';
-import { insertBuild, updateBuild, getBuild, getBuildStatus, getLastRunId } from './db/build';
+import { insertBuild, updateBuild, getBuild, getBuildStatus,
```
Comment (jkuri) [inline src/api/process-manager.ts]: please correct this import as 
```ts
import {
  insertBuild,
  updateBuild,
  getBuild,
  ...
} from './db/build';
```

File: src/api/process-manager.ts [hunk 2]
```diff
@@ -165,7 +166,10 @@ export function startBuild(data: any): Promise<any> {
         repository_id: data.repositories_id,
         data: 'buildAdded'
       });
-    }).catch(err => logger.error(err));
+    })
+    .then(() => getDepracatedBuilds(data.build_id))
+    .then(builds => builds.forEach(b => stopBuild(b)))
```
Comment (jkuri) [inline src/api/process-manager.ts]: let's wait for build to stop, stopping this way won't catch any promise rejections:
```ts
.then(builds => Promise.all(builds.map(build => stopBuild(build)));
```

File: src/api/db/build.ts [hunk 3]
```diff
@@ -282,3 +282,50 @@ export function getBuildStatus(buildId: number): Promise<any> {
...
+export function getDepracatedBuilds(buildId: number): Promise<any> {
+  return new Promise((resolve, reject) => {
+    new Build({ id: buildId }).fetch()
+      .then(build => {
+        if (!build) {
+          reject(build);
+        }
+        build = build.toJSON();
+
+        new Build({ repositories_id: build.repositories_id })
```
Comment (jkuri) [inline src/api/db/build.ts]: is there a reason for running two queries on a `builds` table, you run the query few lines before that already.
let's be rational using resources.

**Comment by Izak88** [comment]
changes after CR...

File: src/api/process-manager.ts [hunk 4]
```diff
@@ -126,46 +133,51 @@ export function startBuild(data: any): Promise<any> {
...
+        jobEvents.next({
+          type: 'process',
+          build_id: data.build_id,
+          repository_id: data.repositories_id,
+          data: 'buildAdded'
         });
-      }, Promise.resolve());
+      })
+      .then(() => getDepracatedBuilds(buildData))
+      .then(builds => Promise.all(builds.map(build => stopBuild(build))));
```
Comment (jkuri) [inline src/api/process-manager.ts]: question: will this force stop all related builds even if not running and write `failed` into database for all that builds?

Comment (Izak88) [reply src/api/process-manager.ts]: I'm preparing bug fix, where all builds gets updated end time, when they are in failed or success state. In that case, those were not returned here, only the ones with end_time null are returned here...

Comment (jkuri) [reply src/api/process-manager.ts]: okay, will this be included in that PR?

Comment (Izak88) [reply src/api/process-manager.ts]: Yes

**Pull request merged**